Prevent confusion about "priority"

okurz opened the pull request:
For the openQA "priority" higher numbers mean actually "less priority"
so we considered different words to use, e.g. "niceness" as described on 
https://en.wikipedia.org/wiki/Nice_%28Unix%29 but is also a very
Unix-specific concept. We might prefer more generic terms. Also such
rename would be a breaking change and would need to provide
backward-compatibility and/or a migration path.

Instead here I opted in documentation, help texts and comments to better
express that the specified value is *only* a value and not the priority
per se which is computed internally based on multiple factors.

This commit also adds a help text for the two -/+ buttons where
previously was no help text at all.

This commit was motivated by recurring user questions about the meaning
and the wish to clarify.

Martchus reviewed (dismissed):
I must say that `niceness` isn't very intuitive either unless you have already the POSIX process setting in mind. I don't see this as a clear improvement, definitely not justifying the huge change.

dcermak commented:
Oliver Kurz ***@***.***> writes:

>> > I see one additional problem with the word "priority" that B would only have real priority over A if all other parameters match, e.g. same worker class.
>> 
>> But is that really related here? Isn't the workerclass used to filter the worker first and then the priority/niceness is used to pick one?
>
> I can tell you from experience that people asked me: "Why is my job not picked up, I already set a priority of -200!!!1"

So now you'll get pings from people saying: "Why is this job not
running, about as un-nice as a $insertUnlikablePerson!!!11eleven" ;-)

okurz commented:
> I'd also like to note that openQA's current behavior is at least consistent with zypper's repository priority.

yeah that explains https://www.reddit.com/r/openSUSE/comments/41bd1h/zypper_priorities_which_numbers_are_highlow/ . Maybe that's why openQA has it the same.

So I tend to keep the word "priority" and not change the logic. But then I hope we can still improve something, at least some help text and documentation? What's a better phrasing than "lower means higher"? ;)

FruitFly638 commented on lib/OpenQA/WebAPI/Controller/API/V1/Job.pm:
```diff
@@ -321,18 +321,18 @@ sub destroy {
     my ($self) = @_;
     return unless my $job = $self->find_job_or_render_not_found($self->stash('jobid'));
-    my $res = $job->set_prio($self->param('prio'));
+    my $res = $job->set_niceness($self->param('niceness'));
```
Does this makes sense though? If we keep `prio` as a fallback, we might as well just not change it at all. It would just make everything more confusing.

okurz commented:
> So I tend to keep the word "priority" and not change the logic. But then I hope we can still improve something, at least some help text and documentation? What's a better phrasing than "lower means higher"? ;)

Anyone has an idea on that?

FruitFly638 commented:
> So I tend to keep the word "priority" and not change the logic. But then I hope we can still improve something, at least some help text and documentation? What's a better phrasing than "lower means higher"? ;)

I think the problem with "lower means higher" is that it is somewhat ambigous. Maybe something like "lower values have higher priority", but that's very clunky.

In UsersGuide it says:
```
The job groups are the place where the actual test scenarios are defined by
the selection of the medium type, the test suite and machine together with a
priority.

The priority is used in the scheduler to choose the next job. If multiple jobs
are scheduled and their requirements for running them are fulfilled the ones
with a lower value for the priority are triggered. The id is the second
sorting key: Of two jobs with equal requirements and same priority the one
with lower id is triggered first.
```
Maybe put the part about lower priority values coming first at the beginning? Right now this part is right in the middle and easy to miss.

okurz commented:
I updated the pull request to revert all renames to "niceness" and opted to only change comments, documentation and help texts to clarify that users merely set a "priority value", not the priority per se.

> I think the problem with "lower means higher" is that it is somewhat ambigous. Maybe something like "lower values have higher priority", but that's very clunky.

Thanks for the ideas. I changed the writing of the texts a little bet to incorporate that while avoiding clunkyness :)

Martchus commented on docs/UsersGuide.asciidoc:
```diff
@@ -982,11 +982,12 @@ _OBSOLETE:: Obsolete jobs in older builds with same DISTRI and VERSION
-_DEPRIORITIZE_LIMIT:: The configurable limit of priority up to which jobs
-should be deprioritized. Needs `_DEPRIORITIZEBUILD`. Default 100.
+_DEPRIORITIZE_LIMIT:: The configurable limit of priority value up to which
+jobs should be deprioritized. Needs `_DEPRIORITIZEBUILD`. Default 100.
```
```suggestion
jobs should be deprioritized. Needs `_DEPRIORITIZEBUILD`. Defaults to 100.
```

okurz commented:
@os-autoinst/tools-team please review for 2nd approval
